orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -491,6 +491,48 @@ IMPLEMENT_GRADIENT_BUILDER(GetConcatGradient) {
               new_attributes)};
 }
 
+IMPLEMENT_GRADIENT_BUILDER(GetConcatTrainingGradient) {
+  auto attributes = SrcNodeAttributes();
+  ORT_ENFORCE(attributes.at("axis").has_i());
```

Comment:
>has_i [](start = 36, length = 5)

i think AttributeProto.type should be used now:
[https://github.com/onnx/onnx/blob/7baca65e03fe9ec819b397fc7a796f4f93999ae8/onnx/onnx.in.proto#L151](https://github.com/onnx/onnx/blob/7baca65e03fe9ec819b397fc7a796f4f93999ae8/onnx/onnx.in.proto#L151)

there's a helper function HasInt(constAttributeProto&):
[https://github.com/microsoft/onnxruntime/blob/1eadec0eea10ad0d0f92ea9df49d67207c8d0564/onnxruntime/core/framework/tensorprotoutils.h#L157](https://github.com/microsoft/onnxruntime/blob/1eadec0eea10ad0d0f92ea9df49d67207c8d0564/onnxruntime/core/framework/tensorprotoutils.h#L157)